include/swift/AST/DiagnosticsSIL.def

```diff
@@ -368,8 +368,9 @@ ERROR(autodiff_unsupported_type,none,
       "differentiating '%0' is not supported yet", (Type))
 ERROR(autodiff_function_not_differentiable,none,
       "function is not differentiable", ())
-NOTE(autodiff_function_generic_functions_unsupported,none,
-     "differentiating generic functions is not supported yet", ())
+NOTE(autodiff_function_indirect_params_or_result_unsupported,none,
```

Comment:
We discussed in person.
I believe "result" is slightly more accurate because the diagnostic is emitted only when the differentiation source result is indirect.
Not changing the diagnostic for now. "Source" will undergo a rehaul soon.